src/OpenTK.Windowing.Desktop/GameWindow.cs

```diff
@@ -195,17 +184,43 @@ public double UpdateFrequency
         public GameWindow(GameWindowSettings gameWindowSettings, NativeWindowSettings nativeWindowSettings)
             : base(nativeWindowSettings)
         {
-            IsMultiThreaded = gameWindowSettings.IsMultiThreaded;
-
-            RenderFrequency = gameWindowSettings.RenderFrequency;
             UpdateFrequency = gameWindowSettings.UpdateFrequency;
         }
 
+        #region Win32 Function for timing
+
+        [DllImport("kernel32", SetLastError = true)]
+        private static extern IntPtr SetThreadAffinityMask(IntPtr hThread, IntPtr dwThreadAffinityMask);
+
+        [DllImport("kernel32")]
+        private static extern IntPtr GetCurrentThread();
+
+        [DllImport("winmm")]
+        private static extern uint timeBeginPeriod(uint uPeriod);
+
+        [DllImport("winmm")]
+        private static extern uint timeEndPeriod(uint uPeriod);
+
+        #endregion
+
+        // Counter for how many updates in Run() where slow.
+        private int _slowUpdates = 0;
+
         /// <summary>
         /// Initialize the update thread (if using a multi-threaded context, and enter the game loop of the GameWindow).
         /// </summary>
         public virtual unsafe void Run()
         {
+            // We do this before OnLoad so that users have some way to affect these settings in OnLoad if they need to.
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                SetThreadAffinityMask(GetCurrentThread(), new IntPtr(1));
```

Comment:
Let's document what this is doing.
"This function call sets the affinity of the current thread to the first CPU. This means that it will not jump between various cores which can cause all sorts of adverse effects."